### PR TITLE
Expose setMetadata

### PR DESCRIPTION
**Pull request by jsani-radar**
https://app.clubhouse.io/radarlabs/story/2241/add-setmetadata-to-js-sdk

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -79,6 +79,19 @@ class Radar {
...
   }
 
+  static setMetadata(metadata) {
+    if (!metadata) {
+      Cookie.deleteCookie(Cookie.METADATA);
+      return;
+    }
+
+    metadata = JSON.stringify(metadata).trim();
+    if (metadata.length === 0 || metadata.length > 256) {
```
Comment (kochis) [inline src/index.js:89]: Why delete if larger than 256?

Comment (jsani-radar) [reply src/index.js:89]: ah, I see - that's the cap on `description` in upsert, `metadata` is capped at 16 keys, changing

Comment (kochis) [reply src/index.js:89]: We also don't need to be doing any validations client-side. It should fail on the server (no need to duplicate the logic here)

Comment (jsani-radar) [reply src/index.js]: also realized it's impossible for `JSON.stringify(metadata)` to have length 0 at this branch with the `if (!metadata)` check above, even stringifying `''` yields a nonzero length
```
> JSON.stringify('');
< """"
```

File: src/index.js [hunk 2]
```diff
@@ -81,6 +81,16 @@ class Radar {
...
   }
 
+  static setMetadata(metadata) {
+    if (!metadata) {
+      Cookie.deleteCookie(Cookie.METADATA);
+      return;
+    }
+
+    metadata = JSON.stringify(metadata);
+    Cookie.setCookie(Cookie.METADATA, metadata);
```
Comment (kochis) [inline src/index.js:91]: Small nit, don't reassign the value:
```js
 Cookie.setCookie(Cookie.METADATA, JSON.stringify(metadata));
```

File: src/index.js [hunk 3]
```diff
@@ -73,19 +73,24 @@ class Radar {
       return;
     }
 
-    description = String(description).trim();
-    if (description.length === 0 || description.length > 256) {
+    const sanitizedDescription = String(description).trim();
+    if (sanitizedDescription.length === 0 || sanitizedDescription.length > 256) {
```
Comment (kochis) [inline src/index.js:77]: I know this is legacy code, but I feel like we should drop the length checks here too. It seems like pretty unexpected behavior to just clear it out.

**Pull request merged**